### PR TITLE
Feature/yeoman generators

### PR DESCRIPTION
**Pull request by ev1stensberg**
Changes the structure to the init feature a bit. We’re now making the parser validate everything, as we can later run the inquirer prompt after running yeoman, reading from a json file. This will allow us to have a more fluid process and removes a lot of «trash» code. Disabled listing for now, as this is work in progress.

This is the groundwork for me to align my work with the AST transformations @pksjce is doing. We're further gonna integrate the prompting from yeoman and we expect to get a config object back from the generator we run.

### CODE REVIEW

File: lib/parser/generators/index.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+const Generator = require('yeoman-generator');
+const {Input} = require('webpack-addons');
+
+class WebpackGenerator extends Generator {
+	constructor(args, opts) {
+		super(args, opts);
```
Comment (okonet) [inline lib/parser/generators/index.js]: No need for constructor in this case.

Comment (ev1stensberg) [inline lib/parser/generators/index.js]: Yeoman needs it, and I'm passing something to it later

File: lib/creator/generators/index.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 const Generator = require('yeoman-generator');
+// eslint-disable-next-line
```
Comment (okonet) [inline lib/creator/generators/index.js]: Why have you disabled the next line? AFAIK destructuring won't work in Node 4.x

File: .yo-rc.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    "config": {
+      "entry": " ",
+      "output": " ",
+      "resolve": " "
+    },
+    "childDependencies": [
+      "webpack-addons-preact"
+    ]
+  }
+}
```
Comment (okonet) [inline .yo-rc.json]: Please add a new line

File: lib/creator/generators/index.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+			output: ' '
+		};
+	}
+	childDependencies() {
+		this.myConfig.childDependencies = ['webpack-addons-preact'];
+	}
+	inject() {}
+}
+
+module.exports = WebpackGenerator;
```
Comment (okonet) [inline lib/creator/generators/index.js]: Why not export directly in the declaration at L5?

File: lib/creator/index.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+* Either throws an error if it doesn't match the webpack schema,
+* or validates the filepaths of the options given.
+* If a package is supplied, it finds the path of the package and runs inquirer
+*
+* @param { Array } pkgPaths - An Array of packages to run
+* @param { <object|null> } opts - An object containing webpackOptions or nothing
+* @returns { <Function|Error> } initTransform - Initializes the scaffold in yeoman
+*/
+
+module.exports = function creator(pkgPaths,opts) {
```
Comment (okonet) [inline lib/creator/index.js]: White space after `,`

File: lib/creator/init-transform.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+		const env = yeoman.createEnv();
+		env.register(require.resolve(options), 'npm:app');
+		env.run('npm:app');
+		try {
+			let name = path.basename(options);
+			console.log('Done!');
+			//eslint-disable-next-line
+		} catch (e) {}
+	}
+	else if(!options) {
```
Comment (okonet) [inline lib/creator/init-transform.js]: This `else if` isn't needed.

File: lib/creator/init-transform.js [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+		const env = yeoman.createEnv();
```
Comment (okonet) [inline lib/creator/init-transform.js]: Can be extracted from here and L18

File: lib/inquirer-prompt.js [hunk 8]
```diff
@@ -9,20 +9,20 @@ const chalk = require('chalk');
...
 * @param { Object } config - Configuration passed from an addon to be included in
 * answers, later sent down to the parser
-* @returns { Function } Parser function that validates the answers, later transforming
+* @returns { Function } Creator function that validates the answers, later transforming
 * the answers to an webpack configuration
 */
 
 module.exports = function prompt(questions, config) {
 	inquirer.prompt(questions).ui.process
 	.reduce(function (newOpts, ans) {
```
Comment (okonet) [inline lib/inquirer-prompt.js]: `function` -> `=>`

File: lib/inquirer-prompt.js [hunk 9]
```diff
@@ -9,20 +9,20 @@ const chalk = require('chalk');
...
 		return attachAnswers(newOpts, ans);
```
Comment (okonet) [inline lib/inquirer-prompt.js]: Actually, `function(newOpts, ans) {...}` -> `attachAnswers`

File: lib/creator/validate-options.spec.js [hunk 10]
```diff
@@ -2,7 +2,7 @@
 
 describe('validate-options', () => {
 	//eslint-disable-next-line
-	const {validateOptions} = require('../../__mocks__/parser/validate-options.mock');
+	const {validateOptions} = require('../../__mocks__/creator/validate-options.mock');
```
Comment (okonet) [inline lib/creator/validate-options.spec.js]: Still eslint-disabled. This won't work in Node 4.x

File: lib/initialize.js [hunk 11]
```diff
@@ -17,7 +14,7 @@ const Rx = require('rx');
 
 module.exports = function initializeInquirer(pkg) {
 	if(pkg.length == 0) {
```
Comment (okonet) [inline lib/initialize.js]: Should we check if `pkg` is `Array`? Otherwise this will be a run-time error.

File: lib/initialize.js [hunk 12]
```diff
@@ -17,7 +14,7 @@ const Rx = require('rx');
...
-		return prompt(Rx.Observable.from(questions), initialConfig);
+		return creator();
 	}
 	else {
```
Comment (okonet) [inline lib/initialize.js]: No need for this `else` here.

File: lib/initialize.js (hunk 11 above)
Comment (ev1stensberg) [inline lib/initialize.js]: Is default in yargs, in the future, when we're revamping the ui, perhaps. This module is gonna be deleted anyway

File: .yo-rc.json (hunk 3 above)
Comment (ev1stensberg) [inline .yo-rc.json]: this is deleted.

**Comment by ev1stensberg** [comment]
@okonet I've removed `inquirer.prompt`, can look on commit history if we need to use it again.

File: lib/initialize.js (hunk 12 above)
Comment (ev1stensberg) [inline lib/initialize.js]: Old code now

**Pull request merged**